velox/functions/prestosql/tests/ComparisonsTest.cpp

```diff
@@ -97,6 +97,72 @@ TEST_F(ComparisonsTest, betweenDate) {
   }
 }
 
+TEST_F(ComparisonsTest, betweenDecimal) {
+  this->options_.parseDecimalAsDouble = false;
+  auto runAndCompare = [&](const std::string& exprStr,
+                           VectorPtr input,
+                           std::vector<bool>& expectedResult) {
+    auto actual = evaluate<SimpleVector<bool>>(exprStr, makeRowVector({input}));
+    for (auto i = 0; i < expectedResult.size(); ++i) {
+      EXPECT_EQ(actual->valueAt(i), expectedResult[i]);
+    }
+  };
+  auto shortFlat = makeNullableShortDecimalFlatVector(
+      {100, 250, 300, 500, std::nullopt}, DECIMAL(3, 2));
+  std::vector<bool> expectedResult{false, true, true, false, false};
+
+  runAndCompare("c0 between 2.00 and 3.00", shortFlat, expectedResult);
+
+  auto longFlat = makeNullableLongDecimalFlatVector(
+      {100, 250, 300, 500, std::nullopt}, DECIMAL(20, 2));
+
+  // Comparing LONG_DECIMAL and SHORT_DECIMAL must throw error.
+  try {
+    runAndCompare("c0 between 2.00 and 3.00", longFlat, expectedResult);
```

Comment:
use `VELOX_ASSERT_THROW` here and below

Reply:
Exception message is quite big for this case. Hence, just comparing a part of the message here.

Reply:
VELOX_ASSERT_THROW will handle part of the message also.
What is the full message? Why is it big?

Reply:
```
```